File: Easy/Card Counting When Easily Distracted/Card Counting When Easily Distracted.cs
using System;
using System.Linq;

class Solution
{
    static void Main(string[] args)
    {   
        char[] cards = new char[] {'A','2','3','4','5','6','7','8','9','T','J','Q','K'};    //Chars, that are valid cards
        string[] s = Console.ReadLine().Split('.');                                         //Create array with all thoughts
        string cardsonly = "";                                                              //This will be populated with valid thoughts
        for (int i = 0; i < s.Length; i++)                                                  //Filter out only card related thoughts
            if (s[i].All(cards.Contains)) cardsonly += s[i];

        string bustThreshold = Console.ReadLine();                                          //Treshhold as string and 
        if (bustThreshold == "10") bustThreshold = "T";                                     //convert to T if 10 

        int belowbust = 0;                                                                  //Will be used to count cards left in the deck
        int abovebust = 0;

        foreach(char i in cards)                                                            //Count each cards in string and
        {                                                                                   // if less than 4 compare to treshhold and add to calc
            if (cardsonly.Count(t => t == i)<4 && Array.IndexOf(cards,i)<Array.IndexOf(cards,Convert.ToChar(bustThreshold)))
            belowbust += 4-cardsonly.Count(t => t == i);
            else if (cardsonly.Count(t => t == i)<4 && Array.IndexOf(cards,i)>=Array.IndexOf(cards,Convert.ToChar(bustThreshold)))
            abovebust += 4-cardsonly.Count(t => t == i);
        }
        Console.WriteLine($"{Math.Round((decimal)belowbust/(belowbust+abovebust)*100)}%");  //Calculate and display result
    }
}

File: Easy/Card Counting When Easily Distracted/Card Counting When Easily Distracted.py
cards = ('A','2','3','4','5','6','7','8','9','T','J','Q','K')
stream_of_consciousness = input().split(".")
cardsonly = ""
for i in stream_of_consciousness:
    if all(c in cards for c in i):
        cardsonly += i

belowbust = 0
abovebust = 0

bust_threshold = input()
if bust_threshold == "10":
    bust_threshold = "T"; 

for char in cards:
    if cardsonly.count(char)<4 and cards.index(char)<cards.index(bust_threshold):
        belowbust += 4-cardsonly.count(char)
    if cardsonly.count(char)<4 and cards.index(char)>=cards.index(bust_threshold):
        abovebust += 4-cardsonly.count(char)

print("%.0f%%" % (belowbust/(belowbust+abovebust)*100))